### PR TITLE
[Feature] Support Stanford Cars dataset.

### PR DESCRIPTION
**Pull request by zzc98**
## Motivation

Support [Stanford Car dataset](https://ai.stanford.edu/~jkrause/cars/car_dataset.html) .

### Example

```
>>> from mmcls.data import StanfordCars

>>> data_root = 'data/stanfordcars'
>>> train=StanfordCars(data_prefix=data_root, test_mode=False, pipeline=[])
>>> len(train)
8144
>>> val = StanfordCars(data_prefix=data_root, test_mode=True, pipeline=[])
>>> len(val)
8041
```

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcls/datasets/stanford_car.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+import numpy as np
+import scipy.io as sio
+
+from .base_dataset import BaseDataset
+from .builder import DATASETS
+
+
+@DATASETS.register_module()
+class StanfordCar(BaseDataset):
+    """The Stanford Car Dataset. Support the `Stanford Car.
```
Comment (Ezra-Yu) [inline mmcls/datasets/stanford_car.py:14]: There are some errors with the format

https://github.com/open-mmlab/mmclassification/blob/d2e505415040bf5329ab218bb6fe3d899f176cd5/mmcls/datasets/mnist.py#L18

File: mmcls/datasets/stanford_car.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+        'Volvo 240 Sedan 1993', 'Volvo XC90 SUV 2007',
+        'smart fortwo Convertible 2012'
+    ]
+
+    def __init__(self,
+                 data_prefix: str,
+                 test_mode: bool,
+                 ann_file: Optional[str] = None,
+                 **kwargs):
+        if test_mode:
```
Comment (Ezra-Yu) [inline mmcls/datasets/stanford_car.py:165]: here `test_mode` is redundant, just use `ann_file` to specificy phase.

File: configs/resnet/resnet50_8xb8_stanford_car.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+_base_ = [
+    '../_base_/models/resnet50.py',
+    '../_base_/datasets/stanford_car_bs8_448.py',
+    '../_base_/schedules/stanford_car_bs8.py', '../_base_/default_runtime.py'
```
Comment (Ezra-Yu) [inline configs/resnet/resnet50_8xb8_stanford_car.py:4]: missing this file '../_base_/schedules/stanford_car_bs8.py'

**Review by Ezra-Yu — APPROVED** [review]
LGTM.

File: requirements/runtime.txt [hunk 4]
```diff
@@ -1,3 +1,4 @@
 matplotlib
 numpy
 packaging
+scipy
```
Comment (mzr1996) [inline requirements/runtime.txt:4]: Please move the `scipy` to the optional requirement file.

File: mmcls/datasets/stanford_car.py [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+            data_prefix=data_prefix,
+            test_mode=test_mode,
+            **kwargs)
+
+    def load_annotations(self):
+        data_infos = []
+        if self.test_mode:
+            data = sio.loadmat(self.test_ann_file)
+        else:
+            data = sio.loadmat(self.train_ann_file)
```
Comment (mzr1996) [inline mmcls/datasets/stanford_car.py:191]: Please import `scipy.io` only when using it. The `scipy` package is a little large, and if we only use it in a dataset class, don't make it a required package.

**Review by mzr1996 — CHANGES_REQUESTED** [review]

File: mmcls/datasets/stanford_car.py (hunk 5 above)
Comment (mzr1996) [reply mmcls/datasets/stanford_car.py:191]: Same in the unit tests.

**Review by mzr1996 — APPROVED** [review]

**Comment by OpenMMLab-Team** [comment]
Hi @zzc98 ！First of all, we want to express our gratitude for your significant PR in the MMClassification project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA

If you have WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution！❤

**Pull request merged**